aws_org_tools/github_oidc/create_github_action_identity_providers.py

```diff
@@ -0,0 +1,209 @@
+            ),
+            Description=f"Role for GitHub Actions deployment of {github_org}/{github_repo}",
+        )
+        print(f"    Created role: {role_name}")
+        print(f"        {response['Role']['Arn']}")
+        return role_name
+    except iam_client.exceptions.EntityAlreadyExistsException:
+        print(f"    Role {role_name} already exists.")
+        response = iam_client.get_role(RoleName=role_name)
+        print(f"        {response['Role']['Arn']}")
+        return role_name
+    except Exception as e:
+        raise (f"Error making role for {github_org}/{github_repo}.") from e
+
+
+def make_assume_role_policy_document(github_oidc_arn, github_org, github_repo):
+    policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": github_oidc_arn,
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringLike": {
+                        "token.actions.githubusercontent.com:sub": f"repo:{github_org}/{github_repo}:*"
+                    },
+                    "StringEquals": {
+                        "token.actions.githubusercontent.com:aud": "sts.amazonaws.com"
+                    },
+                },
+            }
+        ],
+    }
+    return json.dumps(policy)
+
+
+def attach_role_policy(iam_client, role_name):
+    policy_name = "CDK_Deployer_Policy"
+    try:
+        iam_client.put_role_policy(
+            RoleName=role_name,
+            PolicyName=policy_name,
+            PolicyDocument=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Action": ["sts:AssumeRole"],
+                            "Resource": ["arn:aws:iam::*:role/cdk-*"],
+                        }
+                    ],
+                }
+            ),
+        )
+        print(f"    put {policy_name} on role {role_name}")
+    except Exception as e:
+        raise Exception(
+            f"Error putting policy {policy_name} on role {role_name}"
+        ) from e
+
+
+def attach_role_policy_from_file(iam_client, role_name, repo):
+    policy_name = f"{repo.repo_name.replace(' ', '-')}-extra-policy"
+    document = json.load(
+        (Path(__file__).parent / "policies" / repo.policy_file).open()
+    )
+    try:
+        iam_client.put_role_policy(
+            RoleName=role_name,
+            PolicyName=policy_name,
+            PolicyDocument=json.dumps(document),
+        )
+        print(f"    put {policy_name} on role {role_name}")
+    except Exception as e:
+        raise Exception(
+            f"Error putting policy {policy_name} on role {role_name}"
+        ) from e
+
+
+if __name__ == "__main__":
+    configure_github_oicd_provider()
```

Comment:
Here you are calling `configure_github_oicd_provider()` but this function take a param `account`. Shouldn't we be passing it an argument here?

Reply:
I don't think so. 

The way the `@EachAccount` decorator is defined it yields the decorated function with an `AWSAccount` as the first argument. 

So when you call `configure_github_oicd_provider()` it runs through every account in the org, and calls `configure_github_oicd_provider(account)`. This is the `account` object that is then referenced in the function body.

Reply:
got it - ta :+1: